geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -566,10 +566,8 @@ public void addShadowPartitionedRegionForUserPR(PartitionedRegion userPR,
         // Add the overflow statistics to the mbean
         addOverflowStatisticsToMBean(cache, prQ);
 
-        if (!cleanQueues) {
-          // Wait for buckets to be recovered.
-          prQ.shadowPRWaitForBucketRecovery();
-        }
+        // Wait for buckets to be recovered.
+        prQ.shadowPRWaitForBucketRecovery();
```

Comment:
Please add test(s) to ParallelGatewaySenderQueueJUnitTest (the unit test) for this change in addition to any integration tests or distributed tests.